functions/src/firestore.ts

```diff
@@ -0,0 +1,194 @@
+ *   sensor, which is calculated every 10 minutes. Each entry is a map of an
+ *   index in the buffer to an `AqiBufferElement`
+ * - `aqiBufferIndex` - the index of the most recent AQI in the `aqiBuffer`
+ * - `aqiBufferStatus` - if the `aqiBuffer` exists, does not exist, or is in the
+ *   process of being initialized
+ * - `pm25Buffer` - a circular buffer of the last 12 hours of PM2.5 readings for a
+ *   sensor, which is updated every 2 minutes. Each entry is a map of an index
+ *   in the buffer to a `Pm25BufferElement`
+ * - `pm25BufferIndex` - the index of the most recent PM2.5 reading in the `pm25Buffer`
+ * - `pm25BufferStatus` - if the `pm25Buffer` exists, does not exist, or is in the
+ *   process of being initialized
+ * - `lastUpdated` - the last time the sensor doc was updated
+ */
+interface SensorDoc {
+  purpleAirId: number;
+  name: string;
+  latitude: number;
+  longitude: number;
+  isActive: boolean;
+  isValid: boolean;
+  lastSensorReadingTime: FirebaseFirestore.Timestamp | null;
+  lastValidAqiTime: FirebaseFirestore.Timestamp | null;
+  aqiBuffer?: AqiBufferElement[];
+  aqiBufferIndex?: number;
+  aqiBufferStatus: BufferStatus;
+  pm25Buffer?: Pm25BufferElement[];
+  pm25BufferIndex?: number;
+  pm25BufferStatus: BufferStatus;
+  lastUpdated: FirebaseFirestore.Timestamp;
+}
+
+const READINGS = 'readings';
+
+/**
+ * Gets the name of the readings subcollection given the sensor doc ID
+ * @param sensorDocId - Firestore document ID for the sensor
+ * @returns the name of the readings subcollection for a sensor
+ */
+const readingsSubcollection: (sensorDocId: string) => string = (
+  sensorDocId: string
+) => `${SENSORS}/${sensorDocId}/${READINGS}`;
+
+/**
+ * Sensor reading data that is stored in the readings subcollection
+ * - `latitude` - latitude of a sensor
+ * - `longitude` - longitude of a sensor
+ * - `pm25` - PM2.5 reading for a sensor. This value is the average of the
+ *   PM2.5 reading for channelA and channelB
+ * - `humidity` - humidity reading for a sensor
+ * - `meanPercentDifference` - mean percent difference between the pseudo averages
+ *   of the readings for channelA and channelB, as calculated from the confidence
+ *   value returned by the PurpleAir API. This value ranges from 0 to 2.
+ * - `timestamp` - the timestamp of the current reading
+ */
+interface ReadingsDoc {
+  humidity: number;
+  latitude: number;
+  longitude: number;
+  meanPercentDifference: number;
+  pm25: number;
+  timestamp: FirebaseFirestore.Timestamp;
+}
+
+const USERS = 'users';
+
+/**
+ * Data that is stored in a user's doc
+ * - `admin` - if the user is an admin user or not
+ * - `email` - the user's email
+ * - `name` - the user's name, or the empty string if the user has not yet added
+ *   their name
+ */
+interface UserDoc {
+  admin: boolean;
+  email: string;
+  name: string;
+}
+
+const CURRENT_READINGS = 'current-reading';
+
+const SENSORS_DOC = 'sensors';
```

Comment:
`'sensors'` already appears under `SENSORS`. If this is to distinguish the sensors collection from the sensors sub-collection in current-readings, I think it should be called something besides `SENSORS_DOC` since that doesn't really tell me that it's the sub collection.

Reply:
I can append `_COLLECTION` to each of the collection names?